engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EqualsAnyNode.java

```diff
@@ -0,0 +1,574 @@
+  }, limit = "3")
+  boolean equalsTimes(Object selfTime, Object otherTime,
+      @CachedLibrary("selfTime") InteropLibrary selfInterop,
+      @CachedLibrary("otherTime") InteropLibrary otherInterop) {
+    try {
+      return selfInterop.asTime(selfTime).equals(
+          otherInterop.asTime(otherTime)
+      );
+    } catch (UnsupportedMessageException e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  @Specialization(guards = {
+      "selfInterop.isDuration(selfDuration)",
+      "otherInterop.isDuration(otherDuration)"
+  }, limit = "3")
+  boolean equalsDuration(Object selfDuration, Object otherDuration,
+      @CachedLibrary("selfDuration") InteropLibrary selfInterop,
+      @CachedLibrary("otherDuration") InteropLibrary otherInterop) {
+    try {
+      return selfInterop.asDuration(selfDuration).equals(
+          otherInterop.asDuration(otherDuration)
+      );
+    } catch (UnsupportedMessageException e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  /**
+   * Compares interop strings according to the lexicographical order, handling Unicode
+   * normalization. See {@code Text_Utils.compare_to}.
+   */
+  @TruffleBoundary
+  @Specialization(guards = {
+      "selfInterop.isString(selfString)",
+      "otherInterop.isString(otherString)"
+  }, limit = "3")
+  boolean equalsStrings(Object selfString, Object otherString,
+      @CachedLibrary("selfString") InteropLibrary selfInterop,
+      @CachedLibrary("otherString") InteropLibrary otherInterop) {
+    String selfJavaString;
+    String otherJavaString;
+    try {
+      selfJavaString = selfInterop.asString(selfString);
+      otherJavaString = otherInterop.asString(otherString);
+    } catch (UnsupportedMessageException e) {
+      throw new IllegalStateException(e);
+    }
+    return Normalizer.compare(
+        selfJavaString,
+        otherJavaString,
+        Normalizer.FOLD_CASE_DEFAULT
+    ) == 0;
+  }
+
+  @Specialization(guards = {
+      "selfInterop.hasArrayElements(selfArray)",
+      "otherInterop.hasArrayElements(otherArray)"
+  }, limit = "3")
+  boolean equalsArrays(Object selfArray, Object otherArray,
+      @CachedLibrary("selfArray") InteropLibrary selfInterop,
+      @CachedLibrary("otherArray") InteropLibrary otherInterop,
+      @Cached EqualsAnyNode equalsNode
+      ) {
+    try {
+      long selfSize = selfInterop.getArraySize(selfArray);
+      if (selfSize != otherInterop.getArraySize(otherArray)) {
+        return false;
+      }
+      for (long i = 0; i < selfSize; i++) {
+        Object selfElem = selfInterop.readArrayElement(selfArray, i);
+        Object otherElem = otherInterop.readArrayElement(otherArray, i);
+        if (!equalsNode.execute(selfElem, otherElem)) {
+          return false;
+        }
+      }
+      return true;
+    } catch (UnsupportedMessageException | InvalidArrayIndexException e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  /** Equals for Atoms and AtomConstructors */
+
```

Comment:
```suggestion
```

Reply:
Nvm, realised it's a section not a comment to this function. But it was slightly confusing tbh.